fuel-tests/tests/tx.rs

```diff
@@ -121,6 +121,31 @@ async fn submit() {
     assert_eq!(tx.id(), ret_tx.id());
 }
 
+#[tokio::test]
+async fn submit_utxo_verified_tx() {
+    let config = Config {
+        utxo_validation: true,
+        ..Config::local_node()
+    };
+
+    let srv = FuelService::new_node(config).await.unwrap();
+    let client = FuelClient::from(srv.bound_address);
+
+    let tx = Transaction::default();
+
+    let id = client.submit(&tx).await.unwrap();
+    // verify that the tx returned from the api matches the submitted tx
+    let ret_tx = client
+        .transaction(&id.0.to_string())
+        .await
+        .unwrap()
+        .unwrap()
+        .transaction;
+
+    let transaction_result = client.transaction_status(&ret_tx.id().to_string()).await;
+    assert!(transaction_result.is_ok());
```

Comment:
This result indicates whether the client was able to successfully retrieve the status from the server. It doesn't actually interrogate the response to see if the status was actually successful. You'll need to use something like the `matches!` macro to verify the result value is actually successful.

Reply:
Yeah I realized that shortly after I pushed it, almost done with the fix

Reply:
Would you want another check to see `block_id`, and if so what would be the best way to check if a block id is valid vs invalid (I am guessing lookup the block and see if the transaction was included inside of it?)

Reply:
This is pretty good. I thought we'd have to do some genesis coins and transaction signing in the test, but since the fees are zero this works just fine.